WalletWasabi.Fluent/ViewModels/Wallets/Actions/Receive/ReceiveWalletActionViewModel.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Windows.Input;
+using ReactiveUI;
+using WalletWasabi.Blockchain.Analysis.Clustering;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Fluent.ViewModels.NavBar;
+using WalletWasabi.Stores;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets.Actions.Receive
+{
+	[NavigationMetaData(
+		Title = "Receive",
+		Caption = "",
+		IconName = "wallet_action_receive",
+		NavBarPosition = NavBarPosition.None,
+		Searchable = false,
+		NavigationTarget = NavigationTarget.DialogScreen)]
+	public partial class ReceiveWalletActionViewModel : WalletActionViewModel
+	{
+		[AutoNotify] private string _reference;
+		[AutoNotify] private HashSet<string> _suggestions;
+		[AutoNotify] private bool _isExistingAddressesButtonVisible;
+
+		public ReceiveWalletActionViewModel(WalletViewModelBase wallet, WalletManager walletManager, BitcoinStore bitcoinStore) : base(wallet)
+		{
+			Title = "Receive";
+			SelectionMode = NavBarItemSelectionMode.Button;
+			WasabiWallet = wallet.Wallet;
+			_reference = "";
+			_suggestions = GetLabels(walletManager, bitcoinStore);
+
+			var nextCommandCanExecute =
+				this.WhenAnyValue(x => x.Reference)
+					.ObserveOn(RxApp.MainThreadScheduler)
+					.Select(reference => !string.IsNullOrEmpty(reference));
+
+			NextCommand = ReactiveCommand.Create(
+				() =>
+				{
+					var newKey = WasabiWallet.KeyManager.GetNextReceiveKey(Reference, out bool minGapLimitIncreased);
+
+					string? minGapLimitMessage = default;
+					if (minGapLimitIncreased)
+					{
+						int minGapLimit = WasabiWallet.KeyManager.MinGapLimit.Value;
+						int prevMinGapLimit = minGapLimit - 1;
+						minGapLimitMessage = $"Minimum gap limit increased from {prevMinGapLimit} to {minGapLimit}.";
```

Comment:
is this not techno babble? I wonder if a user needs to really know this? or if there is a plain english way to explain it?

Reply:
i doubt many other wallets bother showing this kind of info.

Reply:
Let's say a limit, (for example 50 or something) if the gap limit rise above the limit then give a message to the user like "Too many unused addresses can slow down your wallet." 
IMO bitcoin experts never generate unused address, noobies only generates when they want to receive, this message would only appear in edge cases.

@molnard What is your opinion about this?

Reply:
> "Too many unused addresses can slow down your wallet."

I don't think that is the main reason why that message is displayed. It is to make the user aware of the gap limit they should use when recovering a wallet.